tests/publish-operator.nf

```diff
@@ -0,0 +1,33 @@
+#!/usr/bin/env nextflow
+/*
+ * Copyright 2023, Seqera Labs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+process foo {
+  output:
+    path('output1.txt')
+    path('output2.txt')
+  script:
+    """
+    touch output1.txt
+    touch output2.txt
+    """
+}
+
+workflow {
+  foo()
+  foo.out[0] | publish('.', mode: 'move')
+  foo.out[1] | publish('outputs', mode: 'copy')
```

Comment:
Could be interesting to also support publish on a ChannelOut e.g. `foo | publish`